tests/integration/UploadWordLinkRecordingIntegrationTest.php

```diff
@@ -0,0 +1,228 @@
+
+        # verify exiting recording data isn't changed
+        $this->assertEquals($createRecordingPayload['term'], $createdRecording['term']);
+        $this->assertNotEmpty($createdRecording['fileName']);
+        $this->assertEquals(self::RECORDING_FILE_EXTENSION, pathinfo($createdRecording['fileName'], PATHINFO_EXTENSION));
+        $this->assertEquals(self::TEXT_BACK_TRANSLATION_CONTENT, $createdRecording['textBackTranslation']);
+        $this->assertEquals($this->getProjectId($createRecordingPayload['PhoneId']), $createdRecording['projectId']);
+
+        // verify auto recording is uploaded successfully
+        $uploadedRecordingFile = sprintf("%s/%s/WordLinks/%s", self::$uploadedProjectDir, $createRecordingPayload['PhoneId'], $createdRecording['fileName']);
+        $this->assertFileExists($uploadedRecordingFile);
+        $this->assertEquals(self::AUDIO_FILE_CONTENT, file_get_contents($uploadedRecordingFile));
+    }
+
+    /**
+     * @return void
+     * @throws GuzzleException
+     */
+    public function testShouldNotUpdateGivenRecordingIfPhoneIdIsChanged() {
+        $createRecordingPayload = [
+            'PhoneId' => self::PHONE_ID_1,
+            'term' => self::TERM_2,
+            'wordLinkRecording' => [
+                'audioRecordingFilename' =>self::RECORDING_FILE_NAME_1,
+                'textBackTranslation' => self::TEXT_BACK_TRANSLATION_CONTENT
+            ],
+            'Data' => base64_encode(self::AUDIO_FILE_CONTENT),
+        ];
+
+        $recordingId = $this->sendRequestAndReturnRecordingId($createRecordingPayload);
+        $createdRecording = $this->getRecording($recordingId);
+
+        // request payload
+        $updateRecordingPayload = [
+            'PhoneId' => self::PHONE_ID_2,
+            'term' => self::TERM_2,
+            'wordLinkRecording' => [
+                'audioRecordingFilename' => self::RECORDING_FILE_NAME_2,
+                'textBackTranslation' => self::TEXT_BACK_TRANSLATION_CONTENT_UPDATED
+            ],
+            'Data' => base64_encode(self::AUDIO_FILE_CONTENT_UPDATED),
+            'RecordingId' => $recordingId
+        ];
+
+        // should create a new recording
+        $updatedRecordingId = $this->sendRequestAndReturnRecordingId($updateRecordingPayload);
+        $this->assertNotEquals($createdRecording['id'], $updatedRecordingId);
+
+        # verify exiting recording data isn't changed
+        $this->assertEquals($createRecordingPayload['term'], $createdRecording['term']);
+        $this->assertNotEmpty($createdRecording['fileName']);
+        $this->assertEquals(self::RECORDING_FILE_EXTENSION, pathinfo($createdRecording['fileName'], PATHINFO_EXTENSION));
+        $this->assertEquals(self::TEXT_BACK_TRANSLATION_CONTENT, $createdRecording['textBackTranslation']);
+        $this->assertEquals($this->getProjectId($createRecordingPayload['PhoneId']), $createdRecording['projectId']);
+
+        // verify audio recording is uploaded successfully
```

Comment:
Please also assert `this->getRecording($updatedRecordingId)`.

Reply:
The above block asserts for $recordingId. Please assert for $updatedRecordingId.

Reply:
You're right.